src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +379,285 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                                Stream stream = reader.BaseStream;
+                                stream.Position -= encoding.GetByteCount(sourceLine.Substring(v2Region.StartColumn - 1));
+                            }
+
+                            // Read the next charLength characters
+                            char[] buffer = new char[v2Region.CharLength];
+                            reader.Read(buffer, 0, buffer.Length);
+
+                            byteLength = encoding.GetByteCount(buffer);
+                        }
+                    }
+                    else // Use charOffset
+                    {
+                        // Read the first charOffset characters
+                        char[] buffer = new char[v2Region.CharOffset];
+                        reader.Read(buffer, 0, buffer.Length);
+
+                        // Read the next charLength characters  
+                        buffer = new char[v2Region.CharLength];
+                        reader.Read(buffer, 0, buffer.Length);
+
+                        byteLength = encoding.GetByteCount(buffer);
+                    }
+                }
+            }
+
+            return byteLength;
+        }
+
+        private int GetRegionEndColumn(Region v2Region, Uri uri)
+        {
+            int endColumn = 0;
+            Encoding encoding;
+
+            using (StreamReader reader = GetFileStreamReader(uri, out encoding))
+            {
+                if (reader != null)
+                {
+                    string sourceLine = string.Empty;
+
+                    // Read down to endLine (null return means EOF)
+                    for (int i = 1; i <= v2Region.EndLine && sourceLine != null; sourceLine = reader.ReadLine(), i++) { }
+
+                    if (sourceLine != null)
+                    {
+                        endColumn = sourceLine.Length >= v2Region.EndColumn
+                            ? v2Region.EndColumn
+                            : sourceLine.Length + 1;
+                    }
+                }
+            }
+
+            return endColumn;
+        }
+
+        private Stream GetContentStream(Uri uri, out Encoding encoding)
+        {
+            Stream stream = null;
+            encoding = null;
+            string failureReason = null;
+
+            if (uri != null && _currentV2Run.Files != null)
+            {
+                FileData fileData;
+                if (_currentV2Run.Files.TryGetValue(uri.OriginalString, out fileData))
+                {
+                    // We need the encoding because the content might have been transcoded to UTF-8
+                    string encodingName = fileData.Encoding ?? _currentV2Run.DefaultFileEncoding;
+                    encoding = GetFileEncoding(encodingName);
+
+                    if (encoding != null)
+                    {
+                        if (fileData.Contents != null && fileData.Contents.Binary != null)
```

Comment:
Could be written `if (fileData.Contents?.Binary != null)`.

Reply:
Done